sejda-model/src/main/java/org/sejda/model/validation/validator/SingleOutputValidator.java

```diff
@@ -45,62 +45,21 @@ public void initialize(ValidSingleOutput constraintAnnotation) {
 
     @Override
     public boolean isValid(SingleOutputTaskParameters value, ConstraintValidatorContext context) {
-        if (value != null && value.getOutput() != null) {
-            return isValidOutputType(value);
-        }
-        return true;
-    }
-
-    private boolean isValidOutputType(SingleOutputTaskParameters value) {
-        OutputNameValidatorDispatcher dispatcher = new OutputNameValidatorDispatcher(value.getOutputName());
-        try {
-            value.getOutput().accept(dispatcher);
-        } catch (TaskException e) {
-            // should never happen
-            throw new SejdaRuntimeException(e);
-        }
-        return dispatcher.isValid();
-    }
-
-    /**
-     * Dispatcher to validate according to the runtime type of the task output.
-     * 
-     * @author Andrea Vacondio
-     * 
-     */
-    private static class OutputNameValidatorDispatcher implements TaskOutputDispatcher {
-
-        private boolean valid = true;
-        private String outputName;
-
-        OutputNameValidatorDispatcher(String outputName) {
-            this.outputName = outputName;
-        }
-
-        @Override
-        public void dispatch(FileTaskOutput output) {
-            this.valid = true;
-        }
-
-        @Override
-        public void dispatch(DirectoryTaskOutput output) {
-            this.valid = StringUtils.isNotBlank(outputName);
-
-        }
-
-        @Override
-        public void dispatch(FileOrDirectoryTaskOutput output) {
-            if(output.getDestination().isDirectory()) {
-                this.valid = StringUtils.isNotBlank(outputName);
-            } else {
-                this.valid = true;
+        if (nonNull(value)) {
+            if (isNull(value.getOutput())) {
+                return false;
+            }
+            if (value.getOutput().getDestination().exists()
+                    && (value.getExistingOutputPolicy() == ExistingOutputPolicy.FAIL
+                            || value.getExistingOutputPolicy() == ExistingOutputPolicy.SKIP)) {
```

Comment:
Seems that if value.getOutput().getDestination() is null we crash instead of returning false